### DISCUSSION TITLE
ability to change own attributes?

### DISCUSSION THREAD

**Discussion by markus-96** [ltb-project/self-service-password#746] [Ideas]
I don't know if this is out of scope of this tool, but I don't know any other project covering this amd it fits best here.

The idea: you store phone numbers in your ldap. the phone number of an user changes. the user can update this in ldap. Or: You are doing MAC-based authentication and you want the users to be able to update the list of authorized MACs themselves. 

This feature could have the potential to make this project a full Self Service Portal, not only for passwords.

**Comment by markus-96** [comment]
it could, for example, be configured like this:

```
change_attribute = array(
    array(
        'attribute' => 'telephoneNumber',     # name of the attribute in LDAP
        'who_change' => 'user',               # manager or user change?
        'display_name' => 'Phonenumber',      # how should it be named on the page?
        'multivalued' => false,               # is it a multivalued attribute?
        'read_only => false                   # should the user be able to modify this? could be usefull if you also want to provide some additional information about the user's account
    ),
    array(
        'attribute' => 'sn',
        'display_name' => 'Last Name',
        'multivalued' => false,
        'read_only => true
    ),
    array(
        'attribute' => 'macAdresses',
        'display_name' => 'authorized MAC-Adresses',
        'multivalued' => true,
        'read_only' => false
    )
);
```
what do you think?

I could imagine working on this, but not in the near future.

**Comment by markus-96** [comment]
so something like this in the freeIPA web portal:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/identity_management_guide/self-service

**Comment by coudot** [comment]
We had such feature proposition in White Pages: https://github.com/ltb-project/white-pages/pull/105, that I did not accepted for now.

Fences between Self Service Password, White Pages and Service Desk are thin, but I would like to keep these components for the moment. A global approach would be need to find to best solution.
